apps/app_dial.c

```diff
@@ -2627,7 +2643,20 @@ static int dial_exec_full(struct ast_channel *chan, const char *data, struct ast
 		 */
 		ast_party_connected_line_copy(&tmp->connected, ast_channel_connected(chan));
 
-		topology = ast_stream_topology_clone(ast_channel_get_stream_topology(chan));
+		topology_ds = ast_channel_datastore_find(chan, &topology_ds_info, NULL);
+
+		if (!topology_ds && (topology_ds = ast_datastore_alloc(&topology_ds_info, NULL))) {
```

Comment:
I'm hesitant to incur the extra memory usage in all cases just for the few times someone may want to use the 'j' option. Is there a reason not to put it behind 'j'?

Reply:
It is done deliberately because the channel might go through multiple `Dial` invocations and only the last one might include the `j` option. If we didn't save the topology in the previous invocations, then we would not have access to it.

I understand your point though. Here's an alternative: We do it only for the `j` option and document that for this option to work, the `j` has to be used on all invocations of `Dial`  that the channel goes through.